pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -532,6 +562,14 @@ func (r *Reconciler) ensureIsInstalled(ctx context.Context, log *zap.SugaredLogg
 
 	cmdLog := log.With("cmd", strings.Join(cmd.Args, " "))
 
+	if addon.Name == "csi" && cluster.Spec.Cloud.Hetzner != nil && cluster.Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider] {
```

Comment:
otherwise the addon will not deploy CSIDriver
https://github.com/kubermatic/kubermatic/blob/4782385290c867df50597a7b0670b82478cb7588/addons/csi/hetzner/hcloud-csi.yaml#L22